orbit/pkg/useraction/mdm_migration_darwin.go

```diff
@@ -0,0 +1,293 @@
+//go:build darwin
+
+package useraction
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"sync"
+	"text/template"
+	"time"
+
+	"github.com/rs/zerolog/log"
+)
+
+type swiftDialogExitCode int
+
+const (
+	primaryBtnExitCode   = 0
+	errorExitCode        = 1
+	secondaryBtnExitCode = 2
+	infoBtnExitCode      = 3
+	timeoutExitCode      = 4
+	userQuitExitCode     = 10
+	unknownExitCode      = 99
+)
+
+var mdmMigrationTemplate = template.Must(template.New("mdmMigrationTemplate").Parse(`
+## Migrate to Fleet
+
+To begin, click "Start." Your default browser will open your My Device page.
+
+{{ if .Aggressive }}You {{ else }} Once you start, you {{ end -}} will see this dialog every 15 minutes until you click "Turn on MDM" and complete the instructions.
+
+\\![Image showing the Fleet UI](https://fleetdm.com/images/permanent/mdm-migration-screenshot-768x180@2x.jpg)
+
+Unsure? Contact {{ .OrgInfo.OrgName }} IT [here]({{ .OrgInfo.ContactURL }}).
+`))
+
+var errorTemplate = template.Must(template.New("").Parse(`
+### Something's gone wrong.
+
+Please contact your IT admin [here]({{ .ContactURL }}).
+`))
+
+// baseDialog implements the basic building blocks to render dialogs using
+// swiftDialog.
+type baseDialog struct {
+	path        string
+	interruptCh chan struct{}
+}
+
+func newBaseDialog(path string) *baseDialog {
+	return &baseDialog{path: path, interruptCh: make(chan struct{})}
+}
+
+func (b *baseDialog) CanRun() bool {
+	if _, err := os.Stat(b.path); err != nil {
+		return false
+	}
+
+	return true
+}
+
+// Exit sends the interrupt signal to try and stop the current swiftDialog
+// instance.
+func (b *baseDialog) Exit() {
+	b.interruptCh <- struct{}{}
+	log.Info().Msg("dialog exit message sent")
+}
+
+// render is a general-purpose render method that receives the flags used to
+// display swiftDialog, and starts an asyncronous routine to display the dialog
+// without blocking.
+//
+// The first returned channel sends the exit code returned by swiftDialog, and
+// the second channel is used to send errors.
+func (b *baseDialog) render(flags ...string) (chan swiftDialogExitCode, chan error) {
+	exitCodeCh := make(chan swiftDialogExitCode, 1)
+	errCh := make(chan error, 1)
+	go func() {
+		cmd := exec.Command(b.path, flags...) //nolint:gosec
+		done := make(chan error)
+		stopInterruptCh := make(chan struct{})
+		defer func() { stopInterruptCh <- struct{}{} }()
```

Comment:
Nit: you can do `defer close(stopInterruptCh)` and it will unblock in the `select` (receiving the zero value).